xblock/mixins.py

```diff
@@ -556,9 +556,22 @@ def index_dictionary(self):
         """
         return key/value fields to feed an index within in a Python dict object
         values may be numeric / string or dict
-        default implementation is an empty dict
         """
-        return {}
+        display_name = getattr(self, "display_name", None)
+
+        # Getting self.display_name.default wouldn't work as self.display_name is actually
+        # a str after the class instance is created. So, we can only access the default value
+        # of display_name field by accessing class variable of same name
+        content_type = getattr(
+            getattr(self.__class__, "display_name", None), "default", "Component"
+        )
+
+        return {
+            "content_type": content_type,  # e.g ORA, Video, HTML etc.
+            "content": {
+                "display_name": display_name
+            }
+        }
```

Comment:
I don't think having a default here makes sense. Just don't add these keys to the index if the value is `None`. I don't currently know of a reason why they would be, but still, a default value of "Component", doesn't make sense to me and it also doesn't make sense to index a "None" value.

Reply:
@xitij2000 Fixed